packages/kit/src/runtime/server/data/index.js

```diff
@@ -114,33 +118,45 @@ export async function render_data(
 			)
 		);
 
-		try {
-			const stubs = nodes.slice(0, length).map(serialize_data_node);
+		const chunks = get_data_json(event, options, nodes);
+		const { value: first } = await chunks.next();
 
-			const json = `{"type":"data","nodes":[${stubs.join(',')}]}`;
-			return json_response(json);
-		} catch (e) {
-			const error = /** @type {any} */ (e);
-			return json_response(JSON.stringify(clarify_devalue_error(event, error)), 500);
-		}
+		return new Response(
```

Comment:
Should've added a comment: I didn't always stream because there's an issue we closed some time ago where we added content length headers to the results, and that goes missing when always streaming.

Reply:
ah, fair enough — was just trying to push everything down a single code path. will revert

Reply:
Could we conditionally add a content length header? Then we wouldn't need an extra if path